File: frontend/src/models/Room.js
import { Location } from "./Location";

export class Room {
  constructor(data) {
    this.id = data.id;
    this.name = data.name;
    this.location = data.location ? Location.fromJSON(data.location) : null;
  }

  static fromJSON(json) {
    return new Room(json)
  }
}

File: backend/src/main/java/com/examreg/examreg/models/Room.java
package com.examreg.examreg.models;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Entity
public class Room {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String name;

  @ManyToOne
  @JoinColumn(name = "location_id")
  private Location location;

  @OneToMany(mappedBy = "room")
  private List<ExamSession> examSessions;
}
